### DISCUSSION TITLE
Rewrites + dynamic routes not working in Vercel

### DISCUSSION THREAD

**Discussion by peguerosdc** [vercel/community#7181] [Help]
## Problem

I am trying to deploy a project using Python functions following [this guide](https://vercel.com/docs/functions/runtimes/python).

For the sake of this discussion, consider the following project structure:

```
api
|-index.py
|-app
  |-api
    |-javascript
      |-[slug]
        |-route.ts
```

The following `next.config.mjs`:

```javascript
/** @type {import('next').NextConfig} */
const nextConfig = {
    async rewrites() {
        return [
          {
            source: "/api/python/:path*",
            destination:
              process.env.NODE_ENV === "development"
                ? "http://127.0.0.1:5328/api/python/:path*"
                : "/api/index.py",
          },
        ];
      },
};
export default nextConfig;
```

And the following `index.py`:

```python
from sanic import Sanic
from sanic.request import Request
from sanic.response import json

app = Sanic()

@app.get('/api/python/dummy')
async def index(request: Request):
    return json({"hello": "world"})
```

That is, I want to have two serverless functions:

* One called at `/api/python/dummy`
* One dynamic called at `/api/javascript/[slug]/`

Everything works correctly in my machine, but when I deploy it to Vercel the dynamic route stops working. That is:

* Calling `/api/python/dummy` works.
* Literally calling `/api/javascript/[slug]/` works.
* Calling `/api/javascript/[slug]/` replacing `[slug]` with anything else, doesn't work. What I get is my rendered `404` page instead.

Fun fact: Vercel says all my functions are correctly deployed.

## What I have tried:

1. Renaming the path of `/api/javascript/[slug]/index.ts` to something without the `api` to avoid possible naming conflicts.
2. Using `rewrites` with `beforeFiles`
3. Using `rewrites` with `afterFiles`
4. Using `rewrites` with `fallback`

I have also tried paying for the `pro` account to open a support ticket directly with Vercel, but they haven't replied since Monday and this is a huge blocker for me. 

Has someone seen something similar? [This](https://github.com/orgs/vercel/discussions/6260) is related, but has no replies.

### Example

[nextjs-test-pi-rust.vercel.app](https://nextjs-test-pi-rust.vercel.app/)

### Steps to Reproduce

1. Clone https://github.com/peguerosdc/nextjs-test
2. Deploy in Vercel
3. Click on the 3 buttons in the main page to see the described behavior

**Comment by peguerosdc** [comment]
Ok, so I found this works:

```
api
|-index.py
|-app
  |-other
    |-javascript
      |-[slug]
        |-route.ts
```

With:

* `@app.get('/api/index')` in `index.py`
* No `vercel.json` and no `next.config.mjs`

And the working endpoints are:

* `/api/index`
* `/other/javascript/[slug]`(which works dinamically)

**Reply by APTy** [reply]
@peguerosdc can you post your full working example? it's really not well documented 🙏

[a comment by an automated account is not shown]
